src/main/java/net/snowflake/client/core/QueryContextCache.java

```diff
@@ -137,163 +135,218 @@ public void clearCache() {
   }
 
   /**
-   * Deserialize the query context from the base64 encoded arrow data.
-   *
-   * @param data the base64 encoded query context that was serialized using Arrow.
-   */
-  public void deserializeFromArrowBase64(String data) {
-    synchronized (this) {
-      logger.debug("deserializeFromArrowBase64() called: data from the server = {}", data);
+  * @param data: the QueryContext Object serialized as a JSON format string
+  */
+  public void deserializeQueryContextJson(String data) {
 
+    synchronized (this) {
       // Log existing cache entries
       logCacheEntries();
 
       if (data == null || data.length() == 0) {
         // Clear the cache
         clearCache();
 
-        logger.debug("deserializeFromArrowBase64() returns");
         // Log existing cache entries
         logCacheEntries();
 
         return;
       }
 
-      byte[] decoded = Base64.getDecoder().decode(data);
-      ByteArrayInputStream input = new ByteArrayInputStream(decoded);
-      try (ArrowStreamReader reader = new ArrowStreamReader(input, new RootAllocator())) {
-        while (reader.loadNextBatch()) {
-          VectorSchemaRoot root = reader.getVectorSchemaRoot();
-          for (int i = 0; i < root.getRowCount(); ++i) {
-            QueryContextElement qce = deserializeEntry(root, i);
-            // Merge the element in the existing cache
-            merge(qce.id, qce.readTimestamp, qce.priority, qce.context);
+    try{
+      JsonNode rootNode = jsonObjectMapper.readTree(data);
+
+      // Deserialize the main entry. An example JSON is:
```

Comment:
Can we flatten the query context to a single `queryContext` tag instead of `mainentry` and `entries`. Given the main entry priority and id is always 0 I don't see the need for having two separate tags.

Reply:
I have flattened the query context